prov/util/src/util_ns.c

```diff
@@ -260,23 +260,30 @@ static int util_ns_listen(struct util_ns *ns)
 	if (ret)
 		return -FI_EADDRNOTAVAIL;
 
-	for (p = res; p; p = p->ai_next) {
-		ns->listen_sock = ofi_socket(p->ai_family, p->ai_socktype,
-					     p->ai_protocol);
-		if (ns->listen_sock == INVALID_SOCKET)
-			continue;
-
-		(void) setsockopt(ns->listen_sock, SOL_SOCKET, SO_REUSEADDR,
-				  (void *) &n, sizeof(n));
-		ret = bind(ns->listen_sock, p->ai_addr, (socklen_t) p->ai_addrlen);
-		if (!ret)
-			break;
+	// Linux seems to return IPv4 addresses before IPv6 ones and Windows does
+	// the opposite. So, try IPv4 addresses first followed by IPv6 addresses.
+	for (f = AF_INET; f < AF_INET6; f += (AF_INET6 - AF_INET)) {
```

Comment:
Rather than having this questionable for loop, I would rather pass in the address family into util_ns_listen() and use that as a hint for getaddrinfo().  Then call util_ns_listen() once for each address family if the first call fails.